test/integration/datasources_consul.bats

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bats
+
+load helper
+
+function setup () {
+  tmpdir=$(mktemp -d)
+}
+
+function teardown () {
+  rm -rf $tmpdir
+  consul kv delete foo
+}
+
+@test "Testing consul" {
+  consul kv put foo "$BATS_TEST_DESCRIPTION"
+  gomplate -d consul=consul:// -i '{{(datasource "consul" "foo")}}'
+  [ "$status" -eq 0 ]
+  [[ "${output}" == "$BATS_TEST_DESCRIPTION" ]]
+}
+
+@test "Consul datasource works with hostname in URL" {
+  consul kv put foo "$BATS_TEST_DESCRIPTION"
+  unset CONSUL_HTTP_ADDR
+  gomplate -d consul=consul+http://localhost:8500/ -i '{{(datasource "consul" "foo")}}'
+  [ "$status" -eq 0 ]
+  [[ "${output}" == "$BATS_TEST_DESCRIPTION" ]]
+}
+
+@test "Consul datasource works with consul+http scheme" {
+  consul kv put foo "$BATS_TEST_DESCRIPTION"
+  unset CONSUL_HTTP_ADDR
```

Comment:
oops - this actually won't work (since the addr isn't getting set in the URL on the next line)

It'd probably be best to set `CONSUL_HTTP_ADDR` explicitly in `setup`